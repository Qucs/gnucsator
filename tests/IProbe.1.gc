load bm_wrapper.so
load d_probe.so
load s_sparam.so
load public lang_verilog.so
load c_vpp.so
load c_hide.so
include ../include/sources.v
verilog
param u=3

Vdc #(.U(u)) src(1,0);
R #(.R(1)) res(1,2);
IProbe p1(2,0);

list

spice
.options noinsensitive
.print dc v(nodes) v(p1) i(p1)
.dc

.end
