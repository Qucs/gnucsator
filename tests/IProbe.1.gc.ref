R: already installed, replacing
stashing as R:0
C: already installed, replacing
stashing as C:0
verilog: already installed, replacing
stashing as verilog:0
paramset: already installed, replacing
stashing as paramset:0
module: already installed, replacing
stashing as module:0
macromodule: already installed, replacing
stashing as macromodule:0
verilog: already installed, replacing
stashing as verilog:0
// This File is part of gnucap-qucs 
// (C) 2018 Felix Salfelder 
// GPLv3+ 
// mapping qucsator names to actual devices 
// "sources" section 
//* Vexp:V1 _net4 _net5 U1="0 V" U2="1 V" T1="0" T2="1 ms" Tr="1  ms" Tf="1 ms" 
//* Ipulse:I2 _net2 _net3 I1="0" I2="1 A" T1="0" T2="1 ms" Tr="1 ms" Tf="2 ms" 
//* Irect:I1 _net0 _net1 I="1" TH="1 ms" TL="1 ms" Tr=".5m" Tf=".5m" Td=".5m" 
//* Vac:V1 Gate gnd U="5 V" f="10 MHz" Phase="0" Theta="0" 
//* don't need spice for this 
//* .subckt CCVS(1 2 3 4); 
//* .parameter G=1 
//* .R:0 R1 1 4 1n 
//* H1 2 3 R1 {G} 
//* .ends 
//* .subckt CCCS(1 2 3 4); 
//* .parameter G=1 
//* .R:0 R1 1 4 1n 
//* F1 2 3 R1 {G} 
//* .ends 
//* 
//* only admit in dc and tr 
Vdc #(.U(u),.Temp(NA( NA))) src (.p(1),.n(0));
R #(.R(1),.Tc1(NA( NA)),.Tc2(NA( NA)),.Temp(NA( NA)),.Tnom(NA( NA)),.ic(NA( NA))) res (.p(1),.n(2));
IProbe #() p1 (.p(2),.n(0));
#           v(1)       v(2)       v(p1)      i(p1)      i(p1)     
 0.         3.         29.999u    29.999u    2.9999     2.9999    
