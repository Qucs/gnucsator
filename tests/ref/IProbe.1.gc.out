verilog: already installed, replacing
stashing as verilog:0
paramset: already installed, replacing
stashing as paramset:0
module: already installed, replacing
stashing as module:0
macromodule: already installed, replacing
stashing as macromodule:0
verilog: already installed, replacing
stashing as verilog:0
X: already installed, replacing
stashing as X:0
subckt: already installed, replacing
stashing as subckt:0
clear: already installed, replacing
stashing as clear:0
C: already installed, replacing
stashing as C:0
L: already installed, replacing
stashing as L:0
R: already installed, replacing
stashing as R:0
// This File is part of gnucap-qucs 
// (C) 2018 Felix Salfelder 
// GPLv3+ 
// mapping qucsator names to actual devices 
// "sources" section 
//* Vexp:V1 _net4 _net5 U1="0 V" U2="1 V" T1="0" T2="1 ms" Tr="1  ms" Tf="1 ms" 
//* Ipulse:I2 _net2 _net3 I1="0" I2="1 A" T1="0" T2="1 ms" Tr="1 ms" Tf="2 ms" 
//* Vpulse:V2 _net2 gnd U1="0 V" U2="1 V" T1="5m" T2="10m" Tr="1 ns" Tf="1 ns" 
//* Vrect:V2 _net1 gnd U="1 V" TH="1 ms" TL="1 ms" Tr="1 ns" Tf="1 ns" Td="0 ns" 
//* Irect:I1 _net0 _net1 I="1" TH="1 ms" TL="1 ms" Tr=".5m" Tf=".5m" Td=".5m" 
//* Vac:V1 Gate gnd U="5 V" f="10 MHz" Phase="0" Theta="0" 
//* don't need spice for this 
//* .subckt CCVS(1 2 3 4); 
//* .parameter G=1 
//* .R:0 R1 1 4 1n 
//* H1 2 3 R1 {G} 
//* .ends 
//* .subckt CCCS(1 2 3 4); 
//* .parameter G=1 
//* .R:0 R1 1 4 1n 
//* F1 2 3 R1 {G} 
//* .ends 
//* 
//* only admit in dc and tr 
// This File is part of gnucap-qucs 
// (C) 2018 Felix Salfelder 
// GPLv3+ 
// 
// mapping qucsator names to actual devices 
// "lumped components" section 
// Gyrator:X1 _net0 _net2 gnd gnd R="50 Ohm" Zref="50 Ohm" 
// Zref is some sparam hack.. ignore for now 
// Amp:X1 _net0 _net2 G="10" Z1="50 Ohm" Z2="50 Ohm" NF="0 dB" 
// is it lumped? 
// TLIN:Line1 _net0 _net1 Z="50 Ohm" L="100 mm" Alpha="0 dB" Temp="26.85" 
//                                           is ignored  ^^ 
//* sTr:Tr2 _net4 _net5 gnd _net6 Output Output T1="1" T2="1" 
//* sTr:Tr1 _net0 _net1 gnd _net2 gnd gnd T1="1" T2="1" 
//* MUT:Tr1 _net0 _net1 gnd gnd L1="1 mH" L2="1 mH" k=".9" 
//* MUT2:Tr2 _net5 _net6 _net7 _net8 _net9 _net10 L1="1 mH" L2="1 mH" L3="1 mH" k12="0.9" k13="0.9" k23="0.9" 
//* TODO: check port order 
Vdc #(.U(u)) src (.p(1),.n(0));
R #(.R(1)) res (.p(1),.n(2));
IProbe #() p1 (.p(2),.n(0));
#           v(1)       v(2)       v(p1)      i(p1)      i(p1)     
 0.         3.         29.999u    29.999u    2.9999     2.9999    
